server/command.go

```diff
@@ -125,6 +127,23 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 }
 
 func (p *Plugin) executeLinkCommand(c *plugin.Context, args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	isSysAdmin, err := p.IsSystemAdmin(args.UserId)
```

Comment:
Actually, I think we are already doing that some lines later

Reply:
The permission that is being checked in the code currently is allowed for all channel members in the default permission schema of Mattermost. That's why it's not working and any user is able to link/unlink a channel. Either the default permission schema should be updated in Mattermost server or we should use roles instead of permissions. What do you think @jespino ?

Reply:
I still disagree, that goes against the Mattermost way of doing it. I think probably the permission used is not the right one. I think it should be something like `ManageChannelRoles` would be the right one. Actually, it doesn't need to be distinguished between private or public channels, so it can also simplify a bit the code.